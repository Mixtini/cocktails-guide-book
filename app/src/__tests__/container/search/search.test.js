import React from 'react';
import { render } from '@testing-library/react';
import { toBeInTheDocument } from '@testing-library/jest-dom/matchers';

import Search from '../../../container/search/search';
import SEARCH_TEXT from '../../../assets/wording/search.json';
import SELECTOR from '../../../assets/selector.json';

expect.extend({ toBeInTheDocument });

describe('Test <Search />', () => {
    let search;
    const renderSearch = () => {
        return (
            <Search />
        );
    };
    beforeEach(() => {
        search = render(renderSearch());
    });

    afterEach(() => {
        search.unmount();
    });

    test('Search should display as expected', () => {
        expect(search).toMatchSnapshot();
    });

    test('Title should display as expected', () => {
        const title = search.getByTestId(SELECTOR.SEARCH.TITLE).textContent;
        expect(title).toBe(SEARCH_TEXT.title);
    });
});
