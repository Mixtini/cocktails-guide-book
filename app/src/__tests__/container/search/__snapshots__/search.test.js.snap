// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Search /> Search should display as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzoyAV hNfnvy"
      >
        <div
          class="sc-fzoLag iXuQQf"
          data-testid="TITLE"
        >
          <div>
            今晚喝什麼？
          </div>
        </div>
        <div
          class="sc-fzoXzr bUqkMr"
        >
          <div
            class="sc-fzpjYC eNPMNJ"
            data-testid="CONTENT"
          >
            Sorry! 發生了預期外的錯誤，可能是系統或是網路出了點問題，請稍後再重新整理試試看 😣，如果還是有問題請聯絡我們！謝謝
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzoyAV hNfnvy"
    >
      <div
        class="sc-fzoLag iXuQQf"
        data-testid="TITLE"
      >
        <div>
          今晚喝什麼？
        </div>
      </div>
      <div
        class="sc-fzoXzr bUqkMr"
      >
        <div
          class="sc-fzpjYC eNPMNJ"
          data-testid="CONTENT"
        >
          Sorry! 發生了預期外的錯誤，可能是系統或是網路出了點問題，請稍後再重新整理試試看 😣，如果還是有問題請聯絡我們！謝謝
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
